deps/flatgeobuf/packedrtree.cpp

```diff
@@ -0,0 +1,410 @@
+/******************************************************************************
+ *
+ * Project:  FlatGeobuf
+ * Purpose:  Packed RTree management
+ * Author:   Björn Harrtell <bjorn at wololo dot org>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018-2020, Björn Harrtell <bjorn at wololo dot org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+// NOTE: The upstream of this file is in https://github.com/bjornharrtell/flatgeobuf/tree/master/src/cpp
+
+#ifdef GDAL_COMPILATION
+#include "cpl_port.h"
+#else
+#define CPL_IS_LSB 1
+#endif
+
+#include "packedrtree.h"
+
+#include <map>
+#include <unordered_map>
+#include <iostream>
+
+namespace FlatGeobuf
+{
+
+const NodeItem &NodeItem::expand(const NodeItem &r)
+{
+    if (r.minX < minX) minX = r.minX;
+    if (r.minY < minY) minY = r.minY;
+    if (r.maxX > maxX) maxX = r.maxX;
+    if (r.maxY > maxY) maxY = r.maxY;
+    return *this;
+}
+
+NodeItem NodeItem::create(uint64_t offset)
+{
+    return {
+        std::numeric_limits<double>::infinity(),
+        std::numeric_limits<double>::infinity(),
+        -1 * std::numeric_limits<double>::infinity(),
+        -1 * std::numeric_limits<double>::infinity(),
+        offset
+    };
+}
+
+bool NodeItem::intersects(const NodeItem &r) const
+{
+    if (maxX < r.minX) return false;
+    if (maxY < r.minY) return false;
+    if (minX > r.maxX) return false;
+    if (minY > r.maxY) return false;
+    return true;
+}
+
+std::vector<double> NodeItem::toVector()
+{
+    return std::vector<double> { minX, minY, maxX, maxY };
+}
+
+// Based on public domain code at https://github.com/rawrunprotected/hilbert_curves
+uint32_t hilbert(uint32_t x, uint32_t y)
```

Comment:
64bit output func is available. helps to not bother with prescaling too much.
https://github.com/postgis/postgis/blob/18ce69feba33b13d9fc6122dd594669d9b1bb977/liblwgeom/lwinline.h#L259

Reply:
As mentioned before I wish to keep packedrtree unmodified.